pkg/cmd/cli/cli.go

```diff
@@ -62,43 +62,76 @@ func NewCommandCLI(name, fullName string) *cobra.Command {
 
 	f := clientcmd.New(cmds.PersistentFlags())
 
-	cmds.AddCommand(cmd.NewCmdLogin(fullName, f, in, out))
-	cmds.AddCommand(cmd.NewCmdLogout("logout", fullName+" logout", fullName+" login", f, in, out))
-	cmds.AddCommand(cmd.NewCmdProject(fullName+" project", f, out))
-	cmds.AddCommand(cmd.NewCmdRequestProject("new-project", fullName+" new-project", fullName+" login", fullName+" project", f, out))
-	cmds.AddCommand(cmd.NewCmdNewApplication(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdStatus(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdNewBuild(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdStartBuild(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdCancelBuild(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdBuildLogs(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdDeploy(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdRollback(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdEnv(fullName, f, os.Stdin, out))
-	cmds.AddCommand(cmd.NewCmdExpose(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdGet(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdDescribe(fullName, f, out))
-	// Deprecate 'oc apply' with 'oc create' command.
-	cmds.AddCommand(applyToCreate(cmd.NewCmdCreate(fullName, f, out)))
-	cmds.AddCommand(cmd.NewCmdProcess(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdEdit(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdUpdate(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdDelete(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdLogs(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdImportImage(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdExec(fullName, f, os.Stdin, out, os.Stderr))
-	cmds.AddCommand(cmd.NewCmdPortForward(fullName, f))
-	cmds.AddCommand(cmd.NewCmdProxy(fullName, f, out))
+	groups := templates.CommandGroups{
+		{
+			Message: "Basic Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdLogin(fullName, f, in, out),
+				cmd.NewCmdRequestProject("new-project", fullName+" new-project", fullName+" login", fullName+" project", f, out),
+				cmd.NewCmdNewApplication(fullName, f, out),
+				cmd.NewCmdStatus(fullName, f, out),
+				cmd.NewCmdProject(fullName+" project", f, out),
+			},
+		},
+		{
+			Message: "Build and Deploy Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdStartBuild(fullName, f, out),
+				cmd.NewCmdBuildLogs(fullName, f, out),
+				cmd.NewCmdDeploy(fullName, f, out),
+				cmd.NewCmdRollback(fullName, f, out),
+				cmd.NewCmdNewBuild(fullName, f, out),
+				cmd.NewCmdCancelBuild(fullName, f, out),
+				cmd.NewCmdImportImage(fullName, f, out),
+				cmd.NewCmdScale(fullName, f, out),
+			},
+		},
+		{
+			Message: "Application Modification Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdGet(fullName, f, out),
+				cmd.NewCmdDescribe(fullName, f, out),
```

Comment:
We want people to learn status first, then they get lead into get and describe.

> On Jun 6, 2015, at 4:12 PM, Michail Kargakis <notifications@github.com> wrote:
> 
> In pkg/cmd/cli/cli.go:
> 
> > +			Commands: []*cobra.Command{
> > +				cmd.NewCmdStartBuild(fullName, f, out),
> > +				cmd.NewCmdBuildLogs(fullName, f, out),
> > +				cmd.NewCmdDeploy(fullName, f, out),
> > +				cmd.NewCmdRollback(fullName, f, out),
> > +				cmd.NewCmdNewBuild(fullName, f, out),
> > +				cmd.NewCmdCancelBuild(fullName, f, out),
> > +				cmd.NewCmdImportImage(fullName, f, out),
> > +				cmd.NewCmdScale(fullName, f, out),
> > +			},
> > +		},
> > +		{
> > +			Message: "Application Modification Commands:",
> > +			Commands: []*cobra.Command{
> > +				cmd.NewCmdGet(fullName, f, out),
> > +				cmd.NewCmdDescribe(fullName, f, out),
> Aren't get and describe (status is already there) more suited under the basic commands rather than here? (They don't modify any resources).
> 
> —
> Reply to this email directly or view it on GitHub.
>